mk/config.mk

```
@@ -18,9 +18,12 @@
 
 # Cross-compiler prefix and suffix
 ifeq ($(ARCH),arm)
+# Don't cross-compile if building on aarch64 natively
+ifneq ($(shell arch),aarch64)
 CROSS_COMPILE ?= arm-linux-gnueabihf-
```

Hmm, I didn't try building for an ARM target on ARM64, I guess it really ought to be like this

```suggestion
CROSS_COMPILE ?= arm-linux-gnueabihf-
ifneq ($(shell arch),aarch64)
```

---

Some platforms such as `vexpress-qemu_armv8a` need both a 64-bit and 32-bit compiler because 32-bit TAs are supported as well as 64-bit TAs when the OP-TEE OS is 64-bit. And that's where things get messy, because the `gcc` compiler on an arm64 host certainly cannot produce 32-bit arm. IOW using `gcc` instead of `aarch64-linux-gnu-gcc` is only possible if you're not building the 32-bit TA devkit (SDK).
Still, I am OK with your proposed change. It is small enough and makes your life easier so that's fine :wink:

---

Then yeah it should look like what i propose in the suggestion, since we need an ARM cross compiler on ARM64, but not an ARM64 cross compiler.

Thanks